File: src/test/java/ga/rugal/git/model/GitBlobTest.java
package ga.rugal.git.model;

import java.util.HashSet;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class GitBlobTest {

  @BeforeEach
  public void setUp() {
  }

  @Test
  public void testEquals() {
    final var set = new HashSet<GitBlob>();
    Assertions.assertEquals(0, set.size());
    set.add(new GitBlob("1", "1", 0));
    Assertions.assertEquals(1, set.size());
    set.add(new GitBlob("1", "1", 0));
    Assertions.assertEquals(1, set.size());
    set.add(new GitBlob("1", "3", 0));
    Assertions.assertEquals(1, set.size());
    set.add(new GitBlob("2", "3", 0));
    Assertions.assertEquals(2, set.size());
  }
}

File: library/src/test/kotlin/ga/rugal/gitcleaner/core/model/GitBlobTest.kt
package ga.rugal.gitcleaner.core.model

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.HashSet

class GitBlobTest {

  @Test
  fun test() {
    val set = HashSet<GitBlob>()
    Assertions.assertEquals(0, set.size)
    set.add(GitBlob("1", "1", 0))
    Assertions.assertEquals(1, set.size)
    set.add(GitBlob("1", "1", 0))
    Assertions.assertEquals(1, set.size)
    set.add(GitBlob("1", "3", 0))
    Assertions.assertEquals(1, set.size)
    set.add(GitBlob("2", "3", 0))
    Assertions.assertEquals(2, set.size)
  }
}
